socorro/external/postgresql/connection_context.py

```diff
@@ -2,15 +2,32 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
+import os
 import socket
 import contextlib
 import psycopg2
 import psycopg2.extensions
+from urlparse import urlparse
 
-from configman.config_manager import RequiredConfig
-from configman import Namespace
+from configman import RequiredConfig, Namespace
+from configman.memoize import memoize
 
+#------------------------------------------------------------------------------
+# we're going to end up parsing the same url several times, there's no need to
+# actually do it more than once, so we wrap the method with a memoizer to speed
+# things along.
+@memoize()
```

Comment:
I did that because you complained about rerunning the parser several times in selena's original PR.  I figured you know something about the parsers expense that I did not, I made sure it only ran once.